flashrom: 1.3 -> 1.5.1

r-vdp opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

I cherry-picked the commits from https://github.com/NixOS/nixpkgs/pull/330581 and then added the 1.5.1 update on top.

We can drop the darwin patch since the affected code is only present in the tests and those don't work on darwin anyway, so we can better just don't compile them at all on that platform.

On Linux we do compile and now also run the tests.


<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

edef1c committed:
flashrom: 1.3.0 -> 1.4.0

edef1c committed:
flashrom: build with meson

Per the 1.4.0 release notes, Meson will be the only supported build
system for future versions.

edef1c committed:
flashrom: remove (broken) Promontory support

Release notes for 1.4.0 state that Promontory chipsets are unsupported,
and that attempting to read flash on those systems may crash the system.

The patch that removes this (broken) support only made it into the
1.3.0 release, seemingly by mistake, and the relevant code has been
essentially untouched since.

We'd rather not crash the systems of unsuspecting NixOS/nixpkgs users,
so this change forward-ports the patch from 1.3.0, though amended to
reference the relevant bug in the error message, rather than requesting
the user email upstream.

Link: https://ticket.coreboot.org/issues/370
Link: https://review.coreboot.org/c/flashrom/+/68824

edef1c committed:
flashrom: fix Darwin build

memcpy is a macro on Darwin, and thus requires braced initializer lists
to be parenthesized.

r-vdp commented:
@jiegec I see that you fixed the darwin build in the past. This version bumps breaks it again and I don't really know how to fix it.
Maybe you could take a look if you still care about this package.

prusnak commented:
@r-vdp please have a look at https://github.com/NixOS/nixpkgs/pull/330581 - it seems it contains more fixes than yours PR - it also handles the udev file differently

r-vdp committed:
flashrom: 1.3.0 -> 1.5.1

I cherry-picked the commits from https://github.com/NixOS/nixpkgs/pull/330581
and then added the 1.5.1 update on top.

We can drop the darwin patch since the affected code is only present in
the tests and those don't work on darwin anyway, so we can better just
don't compile them at all on that platform.

On Linux we do compile and now also run the tests.

r-vdp commented:
> @r-vdp please have a look at #330581 - it seems it contains more fixes than yours PR - it also handles the udev file differently

Great, thanks! I don't know why I didn't find that PR.

I cherry-picked the commits from there and added the 1.5.1 bump on top.

r-vdp commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 382143`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>flashrom</li>
    <li>raspberrypi-eeprom</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>flashrom</li>
    <li>raspberrypi-eeprom</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>flashrom</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>flashrom</li>
  </ul>
</details>

r-vdp committed:
flashrom: use finalAttrs

ChocolateLoverRaj reviewed (approved):
I did `nix run nixpkgs#nixpkgs-review -- pr 382143`, and I was able to successfully use `flashrom` with `ch347_spi`. Thank you for this PR.
